node/src/db/db_handlers.rs

```diff
@@ -288,6 +208,69 @@ impl DBHandler {
         }
     }
 }
+pub fn prepare_bead_tuple_data(
+    beads: &Vec<Bead>,
+    bead_index_mapping: &HashMap<BeadHash, usize>,
+    bead: &Bead,
+) -> anyhow::Result<(String, String, String)> {
+    let mut parent_set: HashMap<usize, HashSet<usize>> = HashMap::new();
+
+    for (idx, b) in beads.iter().enumerate() {
```

Comment:
This is incredibly expensive, and in the hot path while the braid lock is being held. 

After 299 you can just use `braid.parents` which has the data you need in O(1).

So I'm going to accept this, but we need to merge #240 then #299 then come back and fix this function.